### PR TITLE
Opr 10 germline sv

### PR DESCRIPTION
**Pull request by LiekeSchoenmaker**
Support all the SV germline events in ROSE/PROTECT/Patient reporter

### CODE REVIEW

File: common/src/main/java/com/hartwig/oncoact/copynumber/ReportableCNVFactory.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    }
+
+    @NotNull
+    public static List<PurpleGainLoss> mergeCNVLists(@NotNull Iterable<PurpleGainLoss> somaticGainsLosses,
+                                                     @Nullable Iterable<PurpleGainLoss> germlineLosses) {
+
+        List<PurpleGainLoss> result = Lists.newArrayList();
+
+        for (PurpleGainLoss somaticCNV : somaticGainsLosses) {
+            result.add(ImmutablePurpleGainLoss.builder()
```
Comment (KasperWolsink99) [inline common/src/main/java/com/hartwig/oncoact/copynumber/ReportableCNVFactory.java:23]: I

File: protect/src/main/java/com/hartwig/oncoact/protect/evidence/VariantEvidence.java [hunk 2]
```diff
@@ -2,9 +2,9 @@
 
 import java.util.Collection;
 import java.util.List;
-import java.util.Set;
 import java.util.stream.Collectors;
 
+import com.google.common.annotations.VisibleForTesting;
```
Comment (KasperWolsink99) [inline protect/src/main/java/com/hartwig/oncoact/protect/evidence/VariantEvidence.java:7]: This import is obsolete

File: common/src/test/java/com/hartwig/oncoact/copynumber/ReportableCNVFactoryTest.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import com.hartwig.hmftools.datamodel.purple.CopyNumberInterpretation;
+import com.hartwig.hmftools.datamodel.purple.PurpleGainLoss;
+import com.hartwig.oncoact.orange.purple.TestPurpleFactory;
+import org.junit.Test;
+
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+public class ReportableCNVFactoryTest {
```
Comment (KasperWolsink99) [inline common/src/test/java/com/hartwig/oncoact/copynumber/ReportableCNVFactoryTest.java:13]: The tests in this class don't really do anything to test the ReportableCNVFactory. (The tests in their current form look like they are focussed on testing the behavior of the Lists::newArrayList method which we don't have to test)

File: common/src/test/java/com/hartwig/oncoact/disruption/ReportableGeneDisruptionFactoryTest.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+import com.google.common.collect.Lists;
+import com.hartwig.hmftools.datamodel.linx.HomozygousDisruption;
+import com.hartwig.oncoact.orange.linx.TestLinxFactory;
+import org.junit.Test;
+
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+public class ReportableGeneDisruptionFactoryTest {
```
Comment (KasperWolsink99) [inline common/src/test/java/com/hartwig/oncoact/disruption/ReportableGeneDisruptionFactoryTest.java:12]: What I said for ReportableCNVFactoryTest also applies to this class

File: common/src/test/java/com/hartwig/oncoact/orange/TestOrangeFactory.java [hunk 5]
```diff
@@ -180,15 +180,40 @@ private static PurpleRecord createTestPurpleRecord() {
...
-        PurpleCopyNumber copyNumber =
+        PurpleVariant germlineVariant = TestPurpleFactory.variantBuilder()
+                .reported(true)
+                .gene("KRAS")
+                .adjustedCopyNumber(6.0)
+                .variantCopyNumber(4.1)
+                .hotspot(Hotspot.HOTSPOT)
+                .subclonalLikelihood(0.02)
+                .biallelic(false)
+                .worstCodingEffect(PurpleCodingEffect.SPLICE)
```
Comment (KasperWolsink99) [inline common/src/test/java/com/hartwig/oncoact/orange/TestOrangeFactory.java:191]: Tiny nitpick but worstCodingEffect, adjustedVAF and repeatCount are already set to the desired values by TestPurpleFactory

File: protect/src/test/java/com/hartwig/oncoact/protect/evidence/CopyNumberEvidenceTest.java [hunk 6]
```diff
@@ -28,33 +28,42 @@ public void canDetermineCopyNumberEvidence() {
         String geneDel = "geneDel";
         ActionableGene amp = TestServeFactory.geneBuilder().gene(geneAmp).event(GeneEvent.AMPLIFICATION).build();
         ActionableGene inactivation = TestServeFactory.geneBuilder().gene(geneDel).event(GeneEvent.INACTIVATION).build();
+        ActionableGene deletion = TestServeFactory.geneBuilder().gene("PTEN").event(GeneEvent.DELETION).build();
```
Comment (KasperWolsink99) [inline protect/src/test/java/com/hartwig/oncoact/protect/evidence/CopyNumberEvidenceTest.java:31]: I would create a new local variable: `String genePten = "PTEN"` to make it more consistent with how it is done for `geneAmp` and `geneDel`

File: protect/src/test/java/com/hartwig/oncoact/protect/evidence/DisruptionEvidenceTest.java [hunk 7]
```diff
@@ -39,20 +39,34 @@ public void canDetermineEvidenceForHomozygousDisruptions() {
...
 
     @NotNull
     private static HomozygousDisruption create(@NotNull String gene) {
         return TestLinxFactory.homozygousDisruptionBuilder().gene(gene).build();
     }
+
+    @NotNull
+    private static ProtectEvidence find(@NotNull List<ProtectEvidence> evidences, @NotNull String geneToFind) {
+        for (ProtectEvidence evidence : evidences) {
+            if (evidence.gene().equals(geneToFind)) {
```
Comment (KasperWolsink99) [inline protect/src/test/java/com/hartwig/oncoact/protect/evidence/DisruptionEvidenceTest.java:65]: evidence.gene() is marked as @Nullable, so I would do a null check

File: protect/src/test/java/com/hartwig/oncoact/protect/evidence/DisruptionEvidenceTest.java [hunk 8]
```diff
@@ -39,20 +39,34 @@ public void canDetermineEvidenceForHomozygousDisruptions() {
...
+
     }
 
     @NotNull
     private static HomozygousDisruption create(@NotNull String gene) {
         return TestLinxFactory.homozygousDisruptionBuilder().gene(gene).build();
     }
+
+    @NotNull
+    private static ProtectEvidence find(@NotNull List<ProtectEvidence> evidences, @NotNull String geneToFind) {
```
Comment (KasperWolsink99) [inline protect/src/test/java/com/hartwig/oncoact/protect/evidence/DisruptionEvidenceTest.java:63]:
```java
        return evidences.stream().filter(x -> Objects.equals(x.gene(), geneToFind))
                .findFirst().orElseThrow(() -> new IllegalStateException("Could not find evidence for gene: " + geneToFind));

```

File: common/src/main/java/com/hartwig/oncoact/copynumber/ReportableCNVFactory.java [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+import java.util.List;
+
+public class ReportableCNVFactory {
+
+    private ReportableCNVFactory() {
+    }
+
+    @NotNull
+    public static List<PurpleGainLoss> mergeCNVLists(@NotNull Iterable<PurpleGainLoss> somaticGainsLosses,
+                                                     @Nullable Iterable<PurpleGainLoss> germlineLosses) {
```
Comment (KasperWolsink99) [inline common/src/main/java/com/hartwig/oncoact/copynumber/ReportableCNVFactory.java:18]: I do not think these merge methods are actually needed. In the end this method is only used to concatenate two lists with each other which can be done inline (see https://stackoverflow.com/questions/189559/how-do-i-join-two-lists-in-java).

I don't think the step where we explicitly convert entries to immutables is needed. We need objects of the PurpleGainLoss type which the elements of these iterables already are.

**Review by KasperWolsink99 — CHANGES_REQUESTED** [review]
Added some comments :) feel free to ask me if you need any additional explanation or help on some comments.

File: protect/src/main/java/com/hartwig/oncoact/protect/evidence/VariantEvidence.java (hunk 2 above)
Comment (LiekeSchoenmaker) [reply protect/src/main/java/com/hartwig/oncoact/protect/evidence/VariantEvidence.java:7]: removed

File: protect/src/test/java/com/hartwig/oncoact/protect/evidence/DisruptionEvidenceTest.java (hunk 7 above)
Comment (LiekeSchoenmaker) [reply protect/src/test/java/com/hartwig/oncoact/protect/evidence/DisruptionEvidenceTest.java:65]: yes correct

File: protect/src/test/java/com/hartwig/oncoact/protect/evidence/DisruptionEvidenceTest.java (hunk 8 above)
Comment (LiekeSchoenmaker) [reply protect/src/test/java/com/hartwig/oncoact/protect/evidence/DisruptionEvidenceTest.java:63]: takes it over, is a nicer solution

File: common/src/test/java/com/hartwig/oncoact/copynumber/ReportableCNVFactoryTest.java (hunk 3 above)
Comment (LiekeSchoenmaker) [reply common/src/test/java/com/hartwig/oncoact/copynumber/ReportableCNVFactoryTest.java]: I was indeed testing if the multiple lists were merging. I have now removed the tests

File: common/src/test/java/com/hartwig/oncoact/disruption/ReportableGeneDisruptionFactoryTest.java (hunk 4 above)
Comment (LiekeSchoenmaker) [reply common/src/test/java/com/hartwig/oncoact/disruption/ReportableGeneDisruptionFactoryTest.java]: same as ReportableCNVFactoryTest

File: common/src/main/java/com/hartwig/oncoact/copynumber/ReportableCNVFactory.java [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+
+import com.google.common.collect.Lists;
+import com.hartwig.hmftools.datamodel.purple.ImmutablePurpleGainLoss;
+import com.hartwig.hmftools.datamodel.purple.PurpleGainLoss;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+
+public class ReportableCNVFactory {
```
[the remark this answers is not in the document]
Comment (LiekeSchoenmaker) [reply common/src/main/java/com/hartwig/oncoact/copynumber/ReportableCNVFactory.java]: What do you suggest? Now it is merging 2 lists, but this class is mean do every logica which is needed for correct interpretation of the data for medical perspective

**Review by mvanniekerkHartwig — APPROVED** [review]
I can not really judge if the "molecular" parts of the code changes are correct. As far as I can tell everything should work. Just one minor suggestion to avoid repeating yourself.

File: common/src/main/java/com/hartwig/oncoact/copynumber/ReportableCNVFactory.java [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+
+import java.util.List;
+
+import com.google.common.collect.Lists;
+import com.hartwig.hmftools.datamodel.purple.PurpleGainLoss;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public final class ReportableCNVFactory {
```
[the remark this answers is not in the document]
Comment (LiekeSchoenmaker) [reply common/src/main/java/com/hartwig/oncoact/copynumber/ReportableCNVFactory.java]: good point :) I have merged them to one class

**Pull request merged**